File: src/main/java/com/lonebytesoft/hamster/accounting/model/Operation.java
package com.lonebytesoft.hamster.accounting.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class Operation implements HasId<Long> {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column
    private long id;

    @Column(name = "transaction_id")
    private long transactionId;

    @Column(name = "account_id")
    private long accountId;

    @Column
    private double amount;

    @Override
    public Long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public long getTransactionId() {
        return transactionId;
    }

    public void setTransactionId(long transactionId) {
        this.transactionId = transactionId;
    }

    public long getAccountId() {
        return accountId;
    }

    public void setAccountId(long accountId) {
        this.accountId = accountId;
    }

    public double getAmount() {
        return amount;
    }

    public void setAmount(double amount) {
        this.amount = amount;
    }

    @Override
    public String toString() {
        return "Operation{" +
                "id=" + id +
                ", transactionId=" + transactionId +
                ", accountId=" + accountId +
                ", amount=" + amount +
                '}';
    }

}

File: src/main/java/com/lonebytesoft/hamster/accounting/model/Operation.kt
package com.lonebytesoft.hamster.accounting.model

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne

@Entity
data class Operation(

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long = 0,

    @ManyToOne(optional = false)
    @JoinColumn(name = "account_id", nullable = false)
    var account: Account = Account(),

    @ManyToOne(optional = true)
    @JoinColumn(name = "currency_id")
    var currency: Currency? = null,

    @Column
    var amount: Double = 0.toDouble(),

    @Column
    var isActive: Boolean = true

) {

    @ManyToOne(optional = false)
    @JoinColumn(name = "transaction_id", nullable = false)
    var transaction: Transaction = Transaction()

}
